src/host/tests/agent.js

```diff
@@ -0,0 +1,185 @@
+/// <reference path="../../data/types.js" />
+
+import test from 'ava';
+
+import { iterateEmitter } from '../../shared/iterate-emitter.js';
+import { startJob } from '../../shared/job.js';
```

Comment:
It's a little unorthodox to use the application logic to express the tests. In this context, the correctness of utilities like `iterateEmitter` and `startJob` is under scrutiny, making it risky to depend on them. If someone were modifying its behavior, then failures here would be especially difficult to interpret. They'd have to determine whether they'd invalidated the application or the test.

Reply:
Hmm. This sounds like if that shared code was versioned and someone maintaining the harness updated to a new major version, introducing a backwards incompatible change. They would need to make that determination as well.

Should I duplicate the relevant mechanics in `iterateEmitter` and `startJob` into this file?

I think the best solution would be to change the patterns for managing this. Though that seems like a large change and I'm not sure what patterns I'd replace them with.

Am I thinking about this in a good direction?